webrecorder/webrecorder/rec/tempchecker.py

```diff
@@ -46,83 +57,91 @@ def delete_if_expired(self, temp_user, temp_dir):
         sesh = self.sesh_redis.get(temp_key)
 
         if sesh == 'commit-wait':
-            try:
-                if not os.path.isdir(temp_dir):
-                    logger.debug('TempChecker: Remove Session For Already Deleted Dir: ' + temp_dir)
-                    self.sesh_redis.delete(temp_key)
-                    return True
-
-                logger.debug('TempChecker: Removing if empty: ' + temp_dir)
-                os.rmdir(temp_dir)
-                #shutil.rmtree(temp_dir)
-                logger.debug('TempChecker: Deleted empty dir: ' + temp_dir)
-
-                self.sesh_redis.delete(temp_key)
+            # This temporary user has signed up for a permanent account and
+            # their collections will be migrated to storage.
+            # Clean up if that migration is complete (i.e. the dir is empty).
+            # Otherwise, wait.
+            if os.path.isdir(temp_dir):
+                try:
+                    logger.debug('TempChecker: Removing if empty: ' + temp_dir)
+                    os.rmdir(temp_dir)
+                    logger.debug('TempChecker: Deleted empty dir: ' + temp_dir)
+                except OSError as e:
+                    if e.errno == 90:
```

Comment:
Wondering if this should be errno 39 for ENOTEMPTY? Curious how 90 happened here, which is ENAMETOOLONG?

Reply:
Wow: I really cannot reconstruct how I got `e.errno == 90`, here or anywhere else. How bizarre, I'm usually so methodical...... closing until I can figure out what I did; will reopen with a correction shortly....